contracts/token/ERC20Receiver.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity ^0.5.0;
+
+import "./ERC20Token.sol";
+
+contract ERC20Receiver {
+
+    event TokenDeposited(address indexed token, address indexed sender, uint256 amount);
+    event TokenWithdrawn(address indexed token, address indexed sender, uint256 amount);
+
+    mapping (address => mapping(address => uint256)) tokenBalances;
+
+    constructor() public {
+        
+    }
+
+    function depositToken(
+        ERC20Token _token
+    )
+        external
+    {
+        _depositToken(
+            msg.sender,
+            _token,
+            _token.allowance(
+                msg.sender, 
+                address(this)
+            )
+        );
+    }
+
+    function withdrawToken(
+        ERC20Token _token,
+        uint256 _amount
+    )
+        external 
+    {
+        _withdrawToken(msg.sender, _token, _amount);
+    }
+
+    function depositToken(
+        ERC20Token _token,
+        uint256 _amount
+    ) 
+        external
+    {
+        require(_token.allowance(msg.sender, address(this)) >= _amount, "Bad argument");
+        _depositToken(msg.sender, _token, _amount);
+    }
+
+    function tokenBalanceOf(
+        ERC20Token _token,
+        address _from
+    )
+        external 
+        view 
+        returns(uint256 fromTokenBalance) 
+    {
+        return tokenBalances[address(_token)][_from];
+    }
+
+    function _depositToken(
+        address _from,
+        ERC20Token _token,
+        uint256 _amount
+    )
+        private 
+    {
+        require(_amount > 0, "Bad argument");
```

Comment:
Just for my information, we deposit negative amounts of tokens?

Reply:
@alaibe , good idea for changing the text to be more explicit.